istioctl/pkg/writer/ztunnel/configdump/api.go

```diff
@@ -0,0 +1,107 @@
+// limitations under the License.
+
+package configdump
+
+type Locality struct {
+	Region  string `json:"region,omitempty"`
+	Zone    string `json:"zone,omitempty"`
+	Subzone string `json:"subzone,omitempty"`
+}
+
+type ZtunnelWorkload struct {
+	WorkloadIPs       []string  `json:"workloadIps"`
+	Waypoint          *Waypoint `json:"waypoint"`
+	Protocol          string    `json:"protocol"`
+	Name              string    `json:"name"`
+	Namespace         string    `json:"namespace"`
+	ServiceAccount    string    `json:"serviceAccount"`
+	WorkloadName      string    `json:"workloadName"`
+	WorkloadType      string    `json:"workloadType"`
+	CanonicalName     string    `json:"canonicalName"`
+	CanonicalRevision string    `json:"canonicalRevision"`
+	ClusterID         string    `json:"clusterId"`
+	TrustDomain       string    `json:"trustDomain,omitempty"`
+	Locality          Locality  `json:"locality,omitempty"`
+	Node              string    `json:"node"`
+	Network           string    `json:"network,omitempty"`
+	Status            string    `json:"status"`
+}
+
+type Waypoint struct {
+	Destination string `json:"destination"`
+}
+
+type LoadBalancer struct {
+	Mode               string   `json:"mode"`
+	RoutingPreferences []string `json:"routingPreferences"`
+}
+
+type ZtunnelService struct {
+	Name         string         `json:"name"`
+	Namespace    string         `json:"namespace"`
+	Hostname     string         `json:"hostname"`
+	Addresses    []string       `json:"vips"`
+	Ports        map[string]int `json:"ports"`
+	LoadBalancer *LoadBalancer  `json:"loadBalancer"`
+	Waypoint     *Waypoint      `json:"waypoint"`
+}
+
+type PolicyMatch struct {
+	Namespaces          []StringMatch `json:"namespaces,omitempty"`
+	NotNamespaces       []StringMatch `json:"notNamespaces,omitempty"`
+	Principals          []StringMatch `json:"principals,omitempty"`
+	NotPrincipals       []StringMatch `json:"notPrincipals,omitempty"`
+	SourceIps           []string      `json:"sourceIps,omitempty"`
+	NotSourceIps        []string      `json:"notSourceIps,omitempty"`
+	DestinationIps      []string      `json:"destinationIps,omitempty"`
+	NotDestinationIps   []string      `json:"notDestinationIps,omitempty"`
+	DestinationPorts    []uint16      `json:"destinationPorts,omitempty"`
+	NotDestinationPorts []uint16      `json:"notDestinationPorts,omitempty"`
+}
+
+type StringMatch struct {
+	Exact    string `json:"Exact,omitempty"`
+	Suffix   string `json:"Suffix,omitempty"`
+	Prefix   string `json:"Prefix,omitempty"`
+	Presence any    `json:"Presence,omitempty"`
+}
+
+type ZtunnelPolicy struct {
+	Name      string             `json:"name"`
+	Namespace string             `json:"namespace"`
+	Scope     string             `json:"scope"`
+	Action    string             `json:"action"`
+	Rules     [][][]*PolicyMatch `json:"rules"`
+}
+
+type ZtunnelDump struct {
+	Workloads    map[string]*ZtunnelWorkload `json:"workloads"`
```

Comment:
The JSON field is modified from `by_addr` to `workloads`, which seems not to be working. Same as services.

Reply:
NVM I see